drake/solvers/MathematicalProgram.cpp

```diff
@@ -107,45 +107,45 @@ class LeastSquares
     : public NonlinearProgram {  // public LinearProgram, public
  public:
   // LinearComplementarityProblem
-  virtual MathematicalProgramInterface* addLinearEqualityConstraint() override {
+  virtual MathematicalProgramInterface* add_linear_equality_constraint() override {
     return new LeastSquares;
   };
-  virtual bool solve(OptimizationProblem& prog) const override {
+  virtual bool Solve(OptimizationProblem& prog) const override {
     size_t num_constraints = 0;
     for (auto const& binding : prog.getLinearEqualityConstraints()) {
-      num_constraints += binding.getConstraint()->getMatrix().rows();
+      num_constraints += binding.get_constraint()->get_matrix().rows();
     }
     
     Eigen::MatrixXd Aeq = Eigen::MatrixXd::Zero(
-        num_constraints, prog.getNumVars());  // todo: use a sparse matrix here?
+        num_constraints, prog.get_num_vars());
     Eigen::VectorXd beq(num_constraints);
     
     size_t constraint_index = 0;
     for (auto const& binding :  prog.getLinearEqualityConstraints()) {
-      auto const& c = binding.getConstraint();
-      size_t n = c->getMatrix().rows();
+      auto const& c = binding.get_constraint();
+      size_t n = c->get_matrix().rows();
       size_t var_index = 0;
-      for (const DecisionVariableView& v : binding.getVariableList()) {
+      for (const DecisionVariableView& v : binding.get_variable_list()) {
         Aeq.block(constraint_index, v.index(), n, v.size()) =
-            c->getMatrix().middleCols(var_index, v.size());
+            c->get_matrix().middleCols(var_index, v.size());
         var_index += v.size();
       }
       beq.segment(constraint_index, n) =
-          c->getLowerBound();  // = c->getUpperBound() since it's an equality
-      // constraint
+          c->get_lower_bound();  // = c->get_upper_bound() since it's an
+                                 // equality constraint
       constraint_index += n;
     }
     
     // least-squares solution
-    prog.setDecisionVariableValues(
+    prog.SetDecisionVariableValues(
         Aeq.jacobiSvd(Eigen::ComputeThinU | Eigen::ComputeThinV).solve(beq));
     return true;
   }
 };
 }
 
-std::shared_ptr<MathematicalProgramInterface> 
-MathematicalProgramInterface::getLeastSquaresProgram() {
+std::shared_ptr<MathematicalProgramInterface>
+MathematicalProgramInterface::get_least_squares_program() {
   return std::shared_ptr<MathematicalProgramInterface>(new LeastSquares);
```

Comment:
Convince me that "you normally wouldn't bother caching its return value when calling it in a loop"?